curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md

```diff
@@ -22,13 +22,13 @@ Declare and initialize a variable <code>total</code> to <code>0</code>. Use a <c
 ```yml
 tests:
   - text: <code>total</code> should be declared and initialized to 0
-    testString: 'assert(code.match(/var.*?total\s*=\s*0.*?;/), "<code>total</code> should be declared and initialized to 0");'
+    testString: 'assert(code.match(/(var|let|const)\s*?total\s*=\s*0.*?;?/), "<code>total</code> should be declared and initialized to 0");'
   - text: <code>total</code> should equal 20
     testString: 'assert(total === 20, "<code>total</code> should equal 20");'
   - text: You should use a <code>for</code> loop to iterate through <code>myArr</code>
     testString: 'assert(code.match(/for\s*\(/g).length > 1 && code.match(/myArr\s*\[/), "You should use a <code>for</code> loop to iterate through <code>myArr</code>");'
   - text: Do not set <code>total</code> to 20 directly
-    testString: 'assert(!code.match(/total[\s\+\-]*=\s*(\d(?!\s*[;,])|[1-9])/g), "Do not set <code>total</code> to 20 directly");'
+    testString: 'assert(!code.match(/total[\s\+\-]*=\s*[^0].*;?/g), "Do not set <code>total</code> to 20 directly");'
```

Comment:
This regex inadvertently allows for setting total to 20 through alternate bases. Changing to "/total[\s\+\-]*=\s*([^0]|0[xbo]?\d+)/g" to fix this.